### ISSUE TITLE
BigQueryTable schema is immutable

### ISSUE THREAD

**Issue by n1koo** [GoogleCloudPlatform/k8s-config-connector#520]
According to [docs](https://cloud.google.com/config-connector/docs/reference/resource-docs/bigquery/bigquerytable) `schema` (of BQ table)  is not immutable but in practice (in latest version) it actually is.

```
Events:
  Type     Reason        Age                  From                      Message
  ----     ------        ----                 ----                      -------
  Warning  UpdateFailed  48s (x525 over 17h)  bigquerytable-controller  Update call failed: cannot make changes to immutable field(s): [schema]
 ```

**Comment by toumorokoshi** [comment]
Hello! Thanks for the error report. 

I took a look, and in this case technically the `schema` field is mutable, but modifications in the schema result in the need to recreate the schema, and thus the changes to the "schema" change require an explicit delete / recreation.

The error message given is incorrect in this case. We'll take a look at improving it.

**Comment by erik-carlson** [comment]
There are cases where updating a schema on an existing table is allowed: https://cloud.google.com/bigquery/docs/managing-table-schemas

It would be nice if config connector supported the cases where an update is allowed, but short of that is there any option rather than abandoning the table KCC resource, updating the schema through another method, and then acquire the table by recreating the table resource with the new schema?

**Comment by krubot-sky** [comment]
Yep this is a big issue for us too, we need to have the cases where config-connector adds new fields allowed.

**Comment by alexguanga** [comment]
@toumorokoshi - Where did you take a look? I'm just curious where in the code is this set? 

> technically the schema field is mutable, but modifications in the schema result in the need to recreate the schema, and thus the changes to the "schema" change require an explicit delete / recreation.

If this is the case, would it make sense that there's at least an option where config connector can do the "refresh"?

**Comment by toumorokoshi** [comment]
> @toumorokoshi - Where did you take a look? I'm just curious where in the code is this set?

Hello! I no longer work on this project, but IIRC this resource uses the google terraform provider under the hood: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/bigquery_table. And I believe in there the schema is immutable.
